### ISSUE TITLE
Generic view model

### ISSUE THREAD

**Issue by ohadnav** [inloop/AndroidViewModel#34]
A quick fix for `ProxyViewHelper` for those of us that failed to create activities or fragments with generic types:
```
public class ProxyViewHelper {
  @Nullable
  public static Class<?> getGenericType(@NonNull Class<?> in, @NonNull Class<?> whichExtends) {
    final Type genericSuperclass = in.getGenericSuperclass();
    if (genericSuperclass instanceof ParameterizedType) {
      final Type[] typeArgs = ((ParameterizedType) genericSuperclass).getActualTypeArguments();
      for (Type arg : typeArgs) {
        if (arg instanceof ParameterizedType) {
          arg = ((ParameterizedType) arg).getRawType();
        }
        if (arg instanceof Class<?>) {
          final Class<?> argClass = (Class<?>) arg;
          if (whichExtends.isAssignableFrom(argClass)) {
            return argClass;
          }
        }
      }
    }
    return null;
  }
}
```

**Comment by DanielNovak** [comment]
@ohadnav Can you post an example generic activity or fragment that is causing issues?

**Comment by ohadnav** [comment]
For example
`ChildFragment extends BaseFragment<ChildModel, BaseViewModel<ChildModel>, FragmentChildBinding>`

**Comment by DanielNovak** [comment]
@ohadnav Thanks for reporting this issue. I have fixed it and version 1.3.3 is ready (on jcenter).
